extensions/wikia/Search/js/WikiaSearch.js

```diff
@@ -81,8 +89,103 @@ var WikiaSearch = {
 			searchForm.submit();
 		});
 
+	},
+	trackSearchResultClick: function(clickedElement) {
+		var queryparams = new URL(window.location).searchParams;
+		var query = queryparams.get('search') || queryparams.get('query');
+
+		if (!query) {
+			return;
+		}
+
+		var payload = {
+			searchPhrase: query,
+			clicked: {
+				type: 'article', // we don't show wikis results right now
+				id: parseInt(clickedElement.getAttribute('data-page-id')),
+				title: clickedElement.text,
+				position: parseInt(clickedElement.getAttribute('data-pos')),
+				thumbnail: !!clickedElement.getAttribute('data-thumbnail'),
+			},
+			target: 'redirect',
+			app: 'app',
+			siteId: parseInt(window.wgCityId),
+			searchId: this.getUniqueSearchId(),
+			pvUniqueId: window.pvUID || "dev", // on dev there is no pvUID available
+		};
+
+		require(['search-tracking', 'wikia.trackingOptIn'], function (searchTracking, trackingOptIn) {
+			trackingOptIn.pushToUserConsentQueue(function () {
+				searchTracking.trackSearchClicked(payload);
+			});
+		});
+	},
+	trackSearchResultsImpression: function() {
+		var queryparams = new URL(window.location).searchParams;
+		var query = queryparams.get('search') || queryparams.get('query');
+
+		if (!query) {
+			return;
+		}
+
+		var results = this.getSearchResults();
+		var payload = {
+			searchPhrase: query,
+			filters: {},
+			results: results,
+			page: parseInt(queryparams.get('page')) || 1,
+			limit: results.length,
+			sortOrder: 'default',
+			app: 'app',
+			siteId: parseInt(window.wgCityId),
+			searchId: this.getUniqueSearchId(),
+			pvUniqueId: window.pvUID || "dev", // on dev there is no pvUID available
+		};
+
+		require(['search-tracking', 'wikia.trackingOptIn'], function (searchTracking, trackingOptIn) {
+			trackingOptIn.pushToUserConsentQueue(function () {
+				searchTracking.trackSearchImpression(payload);
+			});
+		});
+	},
+	getSearchResults: function() {
+		var $results = $('h1 a.result-link[data-page-id]');
+		return $results.map(function(index, item) {
+			return {
+				id: parseInt(item.getAttribute('data-page-id')),
+				title: item.text,
+				position: parseInt(item.getAttribute('data-pos')),
+				thumbnail: !!item.getAttribute('data-thumbnail'),
+			}
+		}).toArray();
+	},
+	getUniqueSearchId: function() {
+		if (this.searchUID) {
+			return this.searchUID;
+		}
+
+		var queryParams = new URL(window.location).searchParams;
+		var searchUID = queryParams.get('searchUID') || this.genUID();
+		this.searchUID = searchUID;
+
+		$('a.paginator-prev, a.paginator-next, a.paginator-page').each(function() {
+			var $elem = $(this);
+			var originalUrl = $elem.attr('href');
+			var modifiedUrl = new URL(originalUrl);
+
+			modifiedUrl.searchParams.append('searchUID', searchUID);
+			$elem.attr('href', modifiedUrl);
+		});
+
+		return searchUID;
+	},
+	genUID: function() {
```

Comment:
We should probably use the shared uuid module you created instead of this, yea?

Reply:
yes, @bkoval is working on sharing it, so we can use it in search suggestions as well